### PR TITLE
edit aggregate pages

### PR DESCRIPTION
**Pull request by adammichaelwood**
addresses #402 
addresses #485 (obliquely -- see notes)

## What is included in this PR?

- Editing for clarity and style on several pages in the Aggregate collection
- Additional prose and images on aggregate-intro
- added the sphinx-tabs extension, for separating OS-specific instructions 
   - this is used in aggregate-app-engine
   - this will be used to solve #485 
- edit images in aggregate-app-engine to remove bookmark links and other browser detritus

## What is left to be done in the addressed issue?

More editing in the Aggregate collection is needed, but I wanted to get the PR merged in because:

 - style and spelling checkers keep improving, and touching files, so keeping working files out longer is a detriment
 - the work so far is a good improvement
 - i feel like i need to move on to the briefcase stuff, and can circle back on Aggregate, which does need work but is generally in better shape

### CODE REVIEW

**Comment by adammichaelwood** [comment]
@yanokwa @ankita240796 

I'm having a problem here. 
There seems to be a conflict between the spelling builder and the tabs extension I added in this PR.
(I am not getting this problem locally, and am not sure how to debug.)

**Comment by ankita240796** [comment]
I think the issue is similar to this one: https://github.com/djungelorm/sphinx-tabs/issues/14#event-1257824316
The spelling builder has to be added to the list of builders for tabs extensions.
https://github.com/djungelorm/sphinx-tabs/commit/22ce7d68407096120ed2061c8d1fd375b8285bcd
We can file an issue in the tabs repo to get this fixed.

**Comment by adammichaelwood** [comment]
I was able to reproduce locally.

I don't know if there's a way to fix the problem directly.
My best suggestion is to not treat warnings as errors (remove `-W` flag). This would allow misspelling to not break the build, so we'd need to do a second check --- catch the misspellings by checking the `output.txt` and raise an error there.

**Comment by ankita240796** [comment]
Yes, that would fix the problem as well. We can generate an error if there is any text in ``output.txt``.

**Comment by adammichaelwood** [comment]
I like your suggestion better, but it depends on getting a patch from another project.

Would you like to file an issue and PR over there?

In the meantime, until we get a patch:
Is it possible to monkeypatch that function in our repo to fix it?
OR
Should we suppress warnings?

**Comment by ankita240796** [comment]
I will file an issue and a PR as well but as of now we can suppress the warnings and use the option to generate an error from ``output.txt`` (until the PR gets merged). I will check if it is possible to monkeypatch the function.

**Comment by adammichaelwood** [comment]
let's not monkeypatch it -- too brittle, I think.

I've suppressed warnings in circleci->spellcheck, so the build passes.
I'll add a quick check on output.txt to fail misspellings.

i'll add

**Comment by adammichaelwood** [comment]
YES! IT FAILED ON PURPOSE!

**Comment by adammichaelwood** [comment]
I seem to have resolved the issues.

If we get a patch from the tabs project, we can remove my ridiculous `check-spelling-output.py` script, but this should work in the mean time.

@yanokwa ready to merge (I hope)

**Comment by yanokwa** [comment]
Can you rebase the commits so we have a cleaner commit history?

**Comment by adammichaelwood** [comment]
I only sort of know what that means in this context. 
Could you be more specific -- like, do you want all these commits collapsed into a single commit?
Or otherwise -- what do you want the end result to look like?

File: src/aggregate-intro.rst [hunk 1]
```diff
@@ -1,14 +1,37 @@
...
-******************
+===================
 
 .. _aggregate-introduction:
 
-ODK Aggregate is a server application and data repository used with :doc:`collect-intro` or other :doc:`OpenRosa-compliant data client applications <openrosa>` to:
+:dfn:`ODK Aggregate` is an open source Java application that
+stores, analyzes, and presents :doc:`XForm <form-design-intro>` survey data collected using :doc:`collect-intro` or other :doc:`OpenRosa-compliant applications <openrosa>`. It supports a wide range of data types, and is designed to work well in any hosting environment.
+
+With Aggregate, survey collection teams can:
```
Comment (yanokwa) [inline src/aggregate-intro.rst]: I usually say data collection. Survey collection reads funny to me.

File: src/aggregate-deployment-planning.rst [hunk 2]
```diff
@@ -88,43 +106,86 @@ You may be able to run a full study within these activity thresholds provided yo
...
+
+.. seealso:: `Learn more about participating in ODK <https://opendatakit.org/participate/>`_
 
-App Engine is usually sufficient
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+.. _aggregate-deployment-app-engine-sufficient:
 
-For most users, Google App Engine will be the easiest and most cost-effective option. The http://opendatakit.appspot.com site, which is lightly used, typically costs just over the minimum charge of $2.10/week unless there are a lot of users that have set up publishing actions (we purge these periodically). 
+.. App Engine is usually sufficient
+   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (yanokwa) [inline src/aggregate-deployment-planning.rst]: Why is this commented out?

**Comment by yanokwa** [comment]
@adammichaelwood Can you clarify the state of the tabs in the rendered page? I'm not sure what I'm supposed to see. A screenshot would help!

File: src/aggregate-deployment-planning.rst (hunk 2 above)
Comment (adammichaelwood) [reply src/aggregate-deployment-planning.rst]: see #525 
This info seemed possibly out of date, so I tried to confirm it.
I couldn't find anything in Google App Engine's current pricing info about weekly billing or weekly minimums.

**Comment by adammichaelwood** [comment]
screen shot 2018-02-21 at 6 02 59 pm
screen shot 2018-02-21 at 6 03 08 pm
screen shot 2018-02-21 at 6 03 15 pm

File: src/aggregate-intro.rst (hunk 1 above)
Comment (adammichaelwood) [reply src/aggregate-intro.rst]: Funny weird or funny haha? :)

**Comment by adammichaelwood** [comment]
btw re: rebasing
I think the appropriate thing is squashing the commits?
Rebasing would still leave a giant stack of weird commits, I think.

**Comment by adammichaelwood** [comment]
I squashed the commits. I think the outcome is about right.

**Comment by ankita240796** [comment]
Sure @yanokwa!

**Comment by ankita240796** [comment]
https://github.com/djungelorm/sphinx-tabs/issues/20
@yanokwa I have filed an issue. I will submit a PR for it as well. Shall a file an issue in the docs repo as well to remove the python script once the PR gets merged?

**Comment by adammichaelwood** [comment]
> For this particular PR, I think rebasing could produce this set of commits that would make it easier for me to review

yeah, I see what you mean.
 
The rebasing tutorial examples don't really cover that level of rearrangement, so it would have taken me some time to wrap my head around presenting it that way, since I went back and forth between a couple different tasks.

So... maybe next time?

**Comment by yanokwa** [comment]
Next time is fine. 

We'll have a tutorial about how to do this during our next developer's call. Sign up https://forum.opendatakit.org/t/odk-1-developer-call-2018-03-07/11699

**Pull request merged**